### PR TITLE
Improve support for ApiGW to StepFunctions service integration

### PR DESCRIPTION
**Pull request by calvernaz**
Issue #7124

https://docs.aws.amazon.com/step-functions/latest/dg/tutorial-api-gateway.html

# Terraform samples

- https://github.com/localstack/localstack-terraform-samples/tree/master/apigateway-step-functions-integration
- https://github.com/localstack/localstack-terraform-samples/tree/master/apigatewayv2-step-functions-integration

# Not covered

- Passthrough templates, for cases where the input is not transformed via request templates (REST) or request parameters (HTTP)

### CODE REVIEW

File: localstack/services/apigateway/integration.py [hunk 1]
```diff
@@ -4,8 +4,9 @@
 from abc import ABC, abstractmethod
 from collections import defaultdict
 from http import HTTPStatus
-from typing import Dict, List, Union
+from typing import Any, Dict, List, Union
 
+from moto.apigatewayv2.exceptions import BadRequestException
```
Comment (whummer) [inline localstack/services/apigateway/integration.py:9]: I think we should import this exception from `localstack.aws.api.apigateway` instead. (to decouple ourselves from `moto` over time)

If we follow the trace of how this exception bubbles up the call stack, we end up at the `ApigatewayRouter`, and our internal middleware/Gateway will be able to handle the ASF generated exception. 👍 

```suggestion
from localstack.aws.api.apigateway import BadRequestException
```

**Review by whummer — COMMENTED** [review]
LGTM in general, great to see more validations, 🚀  and that the StepFunctions integration for `AWS_PROXY` is enhanced.

Looks like the two links to the `localstack-terraform-samples` are not working - have these samples been removed in the meantime?

Wondering if there is an opportunity to add a test for this functionality, to avoid regressions in the future. Perhaps even a unit test would suffice (to test the request parameters creation), or ideally an AWS validated test..

**Comment by calvernaz** [comment]
> LGTM in general, great to see more validations, 🚀 and that the StepFunctions integration for `AWS_PROXY` is enhanced.
> 
> Looks like the two links to the `localstack-terraform-samples` are not working - have these samples been removed/renamed in the meantime?
> 
> Wondering if there is an opportunity to add a test for this functionality, to avoid regressions in the future. Perhaps even a unit test would suffice (to test the request parameters creation), or ideally an AWS validated test..

Yep, I renamed them (updated now). I want to have a more consistent naming going forward.

**Review by whummer — APPROVED** [review]
LGTM @calvernaz  👍

**Pull request merged**